test/util/TickUtils.spec.ts

```diff
@@ -0,0 +1,199 @@
+  ],
+  length: 5,
+  viewBox: { x: 0, y: 0, width: 500, height: 500 },
+  width: 400,
+  x: 0,
+  y: 100,
+};
+
+jest.mock('../../src/util/DOMUtils', () => ({
+  // We mock string size measurement, because getStringSize else returns 0 in these tests.
+  getStringSize: jest.fn(() => ({ width: 20, height: 20 })),
+}));
+
+// These tests have been generated by merely documenting existing behaviour.
+// They will be used to verify that any refactoring does not break the existing behaviour.
+describe('getTicks', () => {
+  describe('ticks are always shown if there is space', () => {
+    test('preserveEnd', () => {
+      const input = { ...EXAMPLE_INPUT, interval: 'preserveEnd' as const };
+
+      const result = getTicks(input);
+
+      expect(result).toEqual([
+        { value: 10, coordinate: 50, tickCoord: 50, isShow: true },
+        { value: 1000, coordinate: 100, tickCoord: 100, isShow: true },
+        { value: 20, coordinate: 150, tickCoord: 150, isShow: true },
+        { value: 40, coordinate: 200, tickCoord: 200, isShow: true },
+        { value: 90, coordinate: 250, tickCoord: 250, isShow: true },
+      ]);
+    });
+
+    test('preserveStart', () => {
+      const input = { ...EXAMPLE_INPUT, interval: 'preserveStart' as const };
+
+      const result = getTicks(input);
+
+      expect(result).toEqual([
+        { value: 10, coordinate: 50, tickCoord: 50, isShow: true },
+        { value: 1000, coordinate: 100, tickCoord: 100, isShow: true },
+        { value: 20, coordinate: 150, tickCoord: 150, isShow: true },
+        { value: 40, coordinate: 200, tickCoord: 200, isShow: true },
+        { value: 90, coordinate: 250, tickCoord: 250, isShow: true },
+      ]);
+    });
+  });
+
+  describe('If not all ticks can be shown, the interval is respected', () => {
+    const viewBoxWithSmallWidth = { x: 0, y: 0, width: 30, height: 500 };
+    test('preserveEnd', () => {
+      const input = {
+        ...EXAMPLE_INPUT,
+        interval: 'preserveEnd' as const,
+        viewBox: viewBoxWithSmallWidth,
+      };
+
+      const result = getTicks(input);
+
+      expect(result).toEqual([
+        {
+          coordinate: 250,
+          isShow: true,
+          tickCoord: 20,
+          value: 90,
+        },
+      ]);
+    });
+
+    test('preserveStart', () => {
+      const input = {
+        ...EXAMPLE_INPUT,
+        interval: 'preserveStart' as const,
+        viewBox: viewBoxWithSmallWidth,
+      };
+
+      const result = getTicks(input);
+
+      expect(result).toEqual([]);
+    });
+
+    test('preserveStartEnd', () => {
+      const input = {
+        ...EXAMPLE_INPUT,
+        interval: 'preserveStartEnd' as const,
+        viewBox: viewBoxWithSmallWidth,
+      };
+
+      const result = getTicks(input);
+
+      expect(result).toEqual([
+        {
+          coordinate: 250,
+          isShow: true,
+          tickCoord: 20,
+          value: 90,
+        },
+      ]);
+    });
+
+    test('0', () => {
```

Comment:
You could consider parametrizing the numeric interval tests or give more meaningful names, like "interval 0 - includes every tick", "interval N - every Nth" and then it might also be worthwhile to write test cases for inputs like _-1_ or when the number is bigger than the number of ticks (and maybe some from other edge cases like NaN, not integers, the maximum number value, infinity etc)

Reply:
+1 for more meaningful names. Else could run these tests in a loop using `each`

Reply:
Good point, done.